lib/publiccloud/utils.pm

```diff
@@ -60,40 +60,81 @@ sub utc_timestamp {
     } elsif (is_sle('<15') && $addon =~ /tcm|wsm|contm|asmm|pcm/) {
-        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '`echo ${VERSION} | cut -d- -f1`', $arch, '', $timeout, $retries, $delay);
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), program => 'SUSEConnect', version => '`echo ${VERSION} | cut -d- -f1`', arch => $arch, params => '', timeout => $timeout, retries => $retries, delay => $delay);
     } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
-        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch, '', $timeout, $retries, $delay);
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), program => 'SUSEConnect', version => '${VERSION_ID}', arch => $arch, params => '', timeout => $timeout, retries => $retries, delay => $delay);
     } else {
         if ($addon =~ /nvidia/i) {
             (my $version = get_version_id(dst_machine => $remote)) =~ s/^(\d+).*/$1/m;
-            ssh_add_suseconnect_product($remote, get_addon_fullname($addon), $version, $arch, '', $timeout, $retries, $delay);
+            ssh_add_suseconnect_product($remote, get_addon_fullname($addon), program => 'SUSEConnect', version => $version, arch => $arch, params => '', timeout => $timeout, retries => $retries, delay => $delay);
         } else {
-            ssh_add_suseconnect_product($remote, get_addon_fullname($addon), undef, $arch, '', $timeout, $retries, $delay);
+            ssh_add_suseconnect_product($remote, get_addon_fullname($addon), program => 'SUSEConnect', version => '${VERSION_ID}', arch => $arch, params => '', timeout => $timeout, retries => $retries, delay => $delay);
         }
     }
     record_info('SUSEConnect time', 'The command SUSEConnect -r ' . get_addon_fullname($addon) . ' took ' . (time() - $cmd_time) . ' seconds.');
 }
 
+=head2 ssh_remove_suseconnect_product
+
+    ssh_remove_suseconnect_product($name, [$version, [$arch, [$params]]]);
+
+Deregister addon in SUT
+=cut
+
+sub ssh_remove_suseconnect_product {
+    my ($remote, $name, $version, $arch, $params) = @_;
+    assert_script_run "sftp $remote:/etc/os-release /tmp/os-release";
+    assert_script_run 'source /tmp/os-release';
+    script_retry(sprintf("ssh $remote sudo SUSEConnect -d -p $name/$version/$arch $params", $remote, $name, $version, $arch, $params), retry => 5, delay => 60, timeout => 180);
+}
+
 sub deregister_addon {
     my ($remote, $addon) = @_;
+
     my $arch = get_var('PUBLIC_CLOUD_ARCH', 'x86_64');
     $arch = "aarch64" if ($arch eq "arm64");
     my $timestamp = utc_timestamp();
     record_info($addon, "Going to deregister '$addon' addon\nUTC: $timestamp");
     my $cmd_time = time();
     my ($timeout, $retries, $delay) = (300, 3, 120);
+
+    assert_script_run "sftp $remote:/etc/os-release /tmp/os-release";
+    assert_script_run 'source /tmp/os-release';
```

Comment:
why do we need to execute `os-release` on the VM in openQA :thinking:

Reply:
Because if you'd do `ssh $remote source /etc/os-autoinst` then the variables would be sourced within that one `ssh` command and won't persist to the next one.

Reply:
was discussed in one on one , make sense now to me